postgres/datadog_checks/postgres/postgres.py

```diff
@@ -72,6 +74,52 @@ def __init__(self, name, init_config, instances):
         self._db_pool = {}
         self._db_pool_lock = threading.Lock()
 
+        self.tags = copy.copy(self._config.tags)
+        self.tags=[t for t in self.tags if not t.startswith("db:")]
+
+        self._dynamic_queries = None
+
+    def _new_query_executor(self, queries):
+        return QueryExecutor(
+            self.execute_query_raw,
+            self,
+            queries=queries,
+            tags=self.tags,
+            hostname=self.resolved_hostname,
+        )
+
+    def execute_query_raw(self, query):
+        with self.db.cursor() as cursor:
+            cursor.execute(query)
+            return cursor.fetchall()
+
+    @property
+    def dynamic_queries(self):
+        if self._dynamic_queries:
+            return self._dynamic_queries
+
+        queries = []
+        if self.version >= V9_2:
+            q_pg_stat_database = copy.deepcopy(QUERY_PG_STAT_DATABASE)
+            q_pg_stat_database["query"] += " WHERE " + " AND ".join(
+                "datname not ilike '{}'".format(db) for db in self._config.ignore_databases
+            )
+
+            if self._config.dbstrict:
+                q_pg_stat_database["query"] += " AND datname in('{}')".format(self._config.dbname)
+
+            queries.extend([q_pg_stat_database])
+
+        if not queries:
+            self.log.debug("no dynamic queries defined")
+            return None
+
+        self._dynamic_queries = self._new_query_executor(queries)
+        self._dynamic_queries.compile_queries()
+        self.log.debug("initialized dynamic queries")
```

Comment:
Not sure this debug statement is super useful on its own but it might be a little more informative if the message included the number of dynamic queries (i.e. `initialized <count> dynamic queries`)

Reply:
I added the count of dynamic queries:

`agent.log:2022-11-29 10:17:37 UTC | CORE | DEBUG | (pkg/collector/python/datadog_agent.go:129 in LogMessage) | postgres:804e6fb9c3022788 | (postgres.py:119) | initialized 1 dynamic querie(s)`

Or do you prefer to remove the line?

Reply:
The updated debug line seems fine to me. Thanks!